windows/src/desktop/kmshell/main/Keyman.System.UpdateStateMachine.pas

```diff
@@ -559,6 +563,16 @@ function TUpdateStateMachine.CurrentStateName: string;
   Result := CurrentState.ClassName;
 end;
 
+function TUpdateStateMachine.ReadyToInstall: Boolean;
+begin
+  if not IsCurrentStateAssigned then
+    Exit(False);
+  if (CurrentState.ClassName = 'WaitingRestartState') and not HasKeymanRun then
```

Comment:
This would be better as:

```suggestion
  if (CurrentState is WaitingRestartState) and not HasKeymanRun then
```

That way we avoid string comparisons which are fragile.

Reply:
> This would be better as:
> 
> That way we avoid string comparisons which are fragile.

missed this but have added in feat/windows/test-online-update